### ISSUE TITLE
Documents Tab on Person Profile Showing Error Page Context Entity Has Not Been Configured

### ISSUE THREAD

**Issue by ChristineBema** [SparkDevNetwork/Rock#5645]
### Description

The core Documents block is occasionally showing this error on version 15.2. You can cause it by going in and changing the Document Types in the block settings. 
![bug 1 docs](https://github.com/SparkDevNetwork/Rock/assets/131175433/d8600488-c517-46c4-98e3-f27b80519f9d)

### Actual Behavior

Sometimes when going to the Documents tab on a person's profile, this message appears: The page context entity has not been configured for this block. Go to Page Properties and click Advanced and enter a valid parameter name under 'Context Parameters', even though the Context Parameter on the page properties is PersonId. This should not throw an error. Sometimes the documents show, like when you clear out all the Document Types allowed in the block and fill them back in, but then the error comes back.

### Expected Behavior

All the documents should show because the Context Parameter on the page, PersonId, is correct. 

### Steps to Reproduce

1. Go to a 15.2 system. Cannot reproduce on demo site.
2. Do to the documents tab on someone's profile. 
3. If the error isn't already there, change the document types allowed.
4. Refresh the page. The error should be there now. 

### Issue Confirmation

- [X] Perform a search on the [Github Issues](https://github.com/issues?q=is%3Aissue+user%3ASparkDevNetwork+-repo%3ARock) to see if your bug or enhancement is already reported.
- [X] Try to reproduce the problem on a fresh install or on the [demo site](https://rock.rocksolidchurchdemo.com/).

### Rock Version

15.2

### Client Culture Setting

en-US

**Comment by leahjennings** [comment]
I'm so glad you submitted this! We are running v16.0 and have this had _intermittently_. It's happened a total of 3x since upgrading. To "fix" it, I will change the document types allowed. The error disappears for a bit, and then someone will report that it's back. Haven't been able to dig into what is triggering it to occur on v16.

**Comment by Koriganp** [comment]
I can also confirm this is occurring on our instance, also 15.2

**Comment by chead4** [comment]
Thank you all for commenting on the issue.  Although I haven't been able to recreate on v15.2 or v16.0 we will continue to investigate.  If the issue occurs again, please include a screenshot that includes the url.

**Comment by ChristineBema** [comment]
The clients I'm seeing it on are on DotLiquid, I don't know if that helps. The URL looks like this: https://rock.pinelake.org/person/132503/persondocs

**Comment by leahjennings** [comment]
We are running Fluid on v16.0.

I just checked, and it "slipped" back to doing it again. Here's a screenshot of the page with the URL I last "fixed" this around 3PM ET today.
<img width="1728" alt="278177316-605ef450-2ee5-48bb-b134-4a9845f9af2c" src="https://github.com/SparkDevNetwork/Rock/assets/66277785/b0e87d68-5fa1-4ea3-a4b6-4a5238b35e07">

Page Properties > Basic Settings:
<img width="1728" alt="image" src="https://github.com/SparkDevNetwork/Rock/assets/66277785/1cd6abdf-a760-4667-8744-62d6f4cffc55">

Page Properties > Display Settings:
<img width="1728" alt="image" src="https://github.com/SparkDevNetwork/Rock/assets/66277785/3653445d-bb0f-4b33-8ceb-aa7c491b1aec">

Page Properties > Advanced Settings:
<img width="1708" alt="image" src="https://github.com/SparkDevNetwork/Rock/assets/66277785/0bd804d6-eb43-4695-9936-30aabaa7b7c5">

Here's a GIF that shows how I "fix" it. I have not yet figured out how to recreate it, I only know when one of our ministers lets me know.
![troubleshootingdocuments](https://github.com/SparkDevNetwork/Rock/assets/66277785/c3ebf57c-cf93-4040-af85-e852ba92d9e7)

**Comment by mijobryco** [comment]
I've experienced the same behavior as above. I removed the Entity Type and Re-added. I am not able to consistently recreate the issue. When adding/removing the Document Types, it appeared to fix the issue but then randomly return. It hasn't however returned since removing and re-adding the Entity Type, at least for me so far.

**Comment by ChristineBema** [comment]
Thanks for that idea! I'll try it and see if it works.

**Comment by mijobryco** [comment]
Just reoccured for me so I guess it isn't the fix, sorry ChristineBema

**Comment by Koriganp** [comment]
I found that doing any change to the block and saving it causes it to work again. (i.e. changing the "Show Security Button" form yes to no, adding or removing one or all Document Types, renaming it). Not sure if that helps.

**Comment by chead4** [comment]
I was able to recreate it today.  Thank you all for the feedback.

**Comment by nairdo** [comment]
**TL;DR**
We believe we've fixed this bug, but please let us know if you see it happen again after updating to v16.1.

**Longer Story**
We were able to capture some data while it was happening on the PreAlpha site and then applied a code change (live, on-the-fly) in order to fix what appeared to be the bug.  (There was a clear condition†† that could cause this particular bug.)  The simplified code change did fix it, and during the live testing we removed the fix and the problem re-appeared.   That's good.  We did that multiple times to verify this.

However, during testing... after adding a document, deleting a document and then putting the code back into the original state, we were _unable_ to re-create the problem (as so many others above have also seen).  Therefore that gives us a small amount of pause, since the state of the things [the "contexts"] seemed to look the same as when the bug was occurring.  Nevertheless, the simplified code _should_ make the condition†† not occur.

**Comment by dataCollegechurch** [comment]
@leahjennings - have you checked to ensure that all of your pages that exist as person profile tabs have PersonId set as a Context parameter. I was experiencing a similar issue in the past and I vaguely remember that this occurred when I was switching between specific tabbed pages.

**Comment by JRichmond27** [comment]
@nairdo are you able to share what the clear condition is that causes this bug? Is it something that's avoidable? 

We're still on 15.3, and this issue is happening very consistently for us. The only fix I've found is to edit the block settings, but then the problem comes back anywhere from a few minutes to a couple hours later.

**Comment by nairdo** [comment]
@JRichmond27 Sorry, I can only give you a technical answer... The condition that would cause this bug is **if** there are no items in the page's "context entity types" collection or if the 'required' context entity types are not in the page's context entity types collection.

The v15.4 alpha testing is almost over, but since the fix above is small enough, I'm going to see if we can get that into v15.4 too.  If we can, we'll also label this issue with "Fixed in v15.4" too.

**Comment by JRichmond27** [comment]
@nairdo that's awesome, thank you!!!
